kong/tools/responses.lua

```diff
@@ -144,6 +143,7 @@ local function send_response(status_code)
         ngx.log(ngx.ERR, "[admin] could not encode value: ", err)
       end
 
+      ngx.header["Content-Type"] = "application/json; charset=utf-8"
```

Comment:
Oh, you moved it here... Better not moving it down here unnecessarily? This instruction is currently voluntarily placed before the response body is produced for cognitive reasons.

Reply:
Do you feel it needs to be added even when:
https://github.com/Kong/kong/blob/f65b9fa5cf463c9f8287fad3b923499d9f69da61/kong/tools/responses.lua#L139

(there is no content)?

Reply:
Yeah, maybe better to keep it in this branch then (but if we hit an error with the JSON-encoding, we might not want to set this header either).

Reply:
Yes, I made it only happen when we are actually outputting someting.